### PR TITLE
[SPARK-9767] Remove ConnectionManager.

### PR DESCRIPTION
**Pull request by rxin**
We introduced the Netty network module for shuffle in Spark 1.2, and has turned it on by default for 3 releases. The old ConnectionManager is difficult to maintain. If we merge the patch now, by the time it is released, it would be 1 yr for which ConnectionManager is off by default. It's time to remove it.

### CODE REVIEW

**Comment by rxin** [comment]
I'm going to merge this since it continuously conflict with other Mima patches. 

cc @tdas to take a quick look at the streaming changes, and @andrewor14 or @JoshRosen to look at rest.

**Comment by tdas** [comment]
LGTM to the streaming change.

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -32,635 +32,638 @@ import com.typesafe.tools.mima.core.ProblemFilters._
...
-              "org.apache.spark.rdd.PairRDDFunctions")
-          ) ++ Seq(
-            // SPARK-4062
-            ProblemFilters.exclude[MissingMethodProblem](
-              "org.apache.spark.streaming.kafka.KafkaReceiver#MessageHandler.this")
-          )
+        // TaskContext was promoted to Abstract class
+        ProblemFilters.exclude[AbstractClassProblem](
+          "org.apache.spark.TaskContext"),
+        Pr
```
Comment (andrewor14) [inline project/MimaExcludes.scala]: line end

**Comment by andrewor14** [comment]
Looks alright.

**Pull request closed without merging**